tree/ntuple/v7/src/RNTupleSerialize.cxx

```diff
@@ -972,24 +989,30 @@ ROOT::Experimental::Internal::RNTupleSerializer::SerializeLocator(const RNTupleL
       throw RException(R__FAIL("locator is not serializable"));
 
    std::uint32_t size = 0;
-   if (locator.fType == RNTupleLocator::kTypeFile) {
-      if (static_cast<std::int32_t>(locator.fBytesOnStorage) < 0)
-         throw RException(R__FAIL("locator too large"));
+   if ((locator.fType == RNTupleLocator::kTypeFile) &&
+       (locator.fBytesOnStorage <= std::numeric_limits<std::int32_t>::max())) {
       size += SerializeUInt32(locator.fBytesOnStorage, buffer);
       size += SerializeUInt64(locator.GetPosition<std::uint64_t>(),
                               buffer ? reinterpret_cast<unsigned char *>(buffer) + size : nullptr);
       return size;
    }
 
+   std::uint8_t locatorType = 0;
    auto payloadp = buffer ? reinterpret_cast<unsigned char *>(buffer) + sizeof(std::int32_t) : nullptr;
    switch (locator.fType) {
-   case RNTupleLocator::kTypeURI: size += SerializeLocatorPayloadURI(locator, payloadp); break;
-   case RNTupleLocator::kTypeDAOS: size += SerializeLocatorPayloadObject64(locator, payloadp); break;
+   case RNTupleLocator::kTypeFile:
+      size += SerializeLocatorPayloadLarge(locator, payloadp);
+      locatorType = 0x01;
```

Comment:
Maybe we should organize these constants in an enum or similar to avoid manually syncing them from Serialize to Deserialize

Reply:
This is actually intentional. The danger with using an enum is that we change the enum (e.g., reorder) and don't realize that this will change the on-disk information.

Reply:
Well, I meant to still explicit the values to avoid reordering issues, like:
```
enum {
  kLocTypeFile = 0x01,
  kLocTypeDAOS = 0x02,
  // ...
};
```
though as long as they are only used in a couple of places in the same file it's not a big deal.